ansible/roles/test/files/ptftests/py3/advanced-reboot.py

```diff
@@ -1358,21 +1367,9 @@ def runTest(self):
                 self.handle_advanced_reboot_health_check_kvm()
                 self.handle_post_reboot_health_check_kvm()
             else:
-                self.handle_advanced_reboot_health_check()
+                self.handle_advanced_reboot_health_check(finalizer_timeout)
                 self.handle_post_reboot_health_check()
 
-            if 'warm-reboot' in self.reboot_type or 'fast-reboot' in self.reboot_type:
```

Comment:
The finalizer check is needed on both KVM and physical tb test. With this change, the check is moved to else condition, and will be skipped for KVM. That's not expected.

Reply:
I've added a check of warmboot_finalizer state before stopping watcher in kvm method as well